app/mails/views.py

```diff
@@ -195,7 +196,9 @@ def mail_update_view(request):
             try:
                 due_id = int(re.sub(r"due-", "", due[0]))
                 d = Due.objects.get(mail=mail, pk=due_id)
-                d.due = due[1][0]
+                d.due = dateparser.parse(
```

Comment:
You'll need to use the date parser for new dues as well (see a few lines above)

Reply:
done

Reply:
done